### PR TITLE
Fix testGatewayPodRestartScenario for ROKS platform

### PR DESCRIPTION
**Pull request by MaxBab**
During execution of testGatewayPodRestartScenario test an active gateway connection tested by using provided gateway hostname.

Currently, the GW node is identified by the nodes which are matching the 'submariner.io/gateway' value.

In ROKS setup, the name of a node from "oc get nodes" with a label of 'submariner.io/gateway' does not match the actual hostname of the GW node.

As a result, the process of locating the active gateway node fails.

In order to fix it, change the process of identifying the active GW node by fetching the "hostname" of the node from the local endpoint of the setup, which provides the correct value for all the platforms.

### CODE REVIEW

File: test/e2e/redundancy/gateway_failover.go [hunk 1]
```diff
@@ -87,7 +87,8 @@ func testGatewayPodRestartScenario(f *subFramework.Framework) {
 
 	framework.By(fmt.Sprintf("Ensuring that the gateway reports as active on %q", primaryClusterName))
 
-	activeGateway := f.AwaitGatewayFullyConnected(primaryCluster, resource.EnsureValidName(gatewayNodes[0].Name))
```
Comment (yboaron) [inline test/e2e/redundancy/gateway_failover.go:90]: Maybe leave a comment here to explain why hostname is retrieved from endpoint instead of gatewayNodes[0].Name

**Review by yboaron — APPROVED** [review]

File: test/e2e/redundancy/gateway_failover.go [hunk 2]
```diff
@@ -87,7 +87,8 @@ func testGatewayPodRestartScenario(f *subFramework.Framework) {
...
+	submEndpoint := f.AwaitSubmarinerEndpoint(primaryCluster, subFramework.NoopCheckEndpoint)
```
Comment (yboaron) [inline test/e2e/redundancy/gateway_failover.go:90]: It is corner case, but it better check that  submEndpoint  is not nil

Expect(submEndpoint).ToNot(BeNil(), "Did not find local endpoint")

Comment (tpantelis) [reply test/e2e/redundancy/gateway_failover.go:90]: It's actually not necessary b/c AwaitSubmarinerEndpoint will itself fail the test if it's not found. Also note that other call sites don't check for nil.

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Comment by dfarrell07** [comment]
@MaxBab I think this should be backported?

**Comment by MaxBab** [comment]
@dfarrell07 
This should be backported to release-0.18 branch.
I'll do it.
